app/src/main/java/io/homeassistant/companion/android/settings/qs/views/ManageTilesView.kt

```diff
@@ -0,0 +1,144 @@
+            for ((tileName, tileId) in tileNameArray.zip(tileIdArray)) {
+                DropdownMenuItem(onClick = {
+                    viewModel.selectedTile.value = tileId
+                    expandedTile = false
+                    if (viewModel.currentTile() != null) {
+                        viewModel.tileLabel.value = viewModel.currentTile()!!.label
+                        viewModel.tileSubtitle.value = viewModel.currentTile()!!.subtitle
+                        viewModel.selectedEntityId.value = viewModel.currentTile()!!.entityId
+                        viewModel.selectedIcon.value = viewModel.currentTile()!!.iconId
+                        val loader = IconPackLoader(context)
+                        iconPack = createMaterialDesignIconPack(loader)
+                        iconPack.loadDrawables(loader.drawableLoader)
+                        val iconDrawable = viewModel.selectedIcon.value?.let { iconPack.getIcon(it)?.drawable }
+                        if (iconDrawable != null) {
+                            val icon = DrawableCompat.wrap(iconDrawable)
+                            icon.setColorFilter(context.resources.getColor(io.homeassistant.companion.android.R.color.colorAccent), PorterDuff.Mode.SRC_IN)
+                            viewModel.icon.value = icon
+                        }
+                    }
+                }) {
+                    Text(tileName)
+                }
+            }
+        }
+        Text(
+            text = stringResource(id = R.string.tile_label),
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        TextField(
+            value = viewModel.tileLabel.value.toString(),
+            onValueChange = { viewModel.tileLabel.value = it },
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        Text(
+            text = stringResource(id = R.string.tile_subtitle),
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        TextField(
+            value = viewModel.tileSubtitle.value.toString(),
+            onValueChange = { viewModel.tileSubtitle.value = it },
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+
+        TextButton(onClick = {
+            iconDialog.show(childFragment, viewModel.selectedTile.value)
+        }) {
+            Text(
+                text = stringResource(id = R.string.tile_icon),
+                fontSize = 15.sp,
+                modifier = Modifier.padding(end = 10.dp)
+            )
+            viewModel.icon.value?.toBitmap()?.asImageBitmap()
+                ?.let { Image(it, contentDescription = stringResource(id = R.string.tile_icon)) }
+        }
+        TextButton(onClick = { expandedEntity = true }) {
+            Text(
+                text = stringResource(id = R.string.tile_entity) + ": ${viewModel.selectedEntityId.value}",

```

Comment:
If no entity has been selected this will show up as "Select a entity to toggle or call: null`, it should probably be changed to show "not set" or something similar if the value is null

Reply:
actually a user is required to supply a value so I will opt to show a blank string, we can't add a tile without a label and entity ID at a minimum and clearly I missed this during the migration 😬 I will also update the save button to reflect this.